sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -796,3 +797,64 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function which returns all the keys of outer JSON object.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(json_object) - returns all the keys of outer JSON object.",
+  arguments = """
+    Arguments:
+      * json_object - A JSON object. If it is an invalid string, the function returns null.
+          If it is a JSON array or null, a runtime exception will be thrown.
```

Comment:
Is there a reason why you choose a runtime exception for `null`? In SQL world, for `null` input, `null` return is expected instead of a runtime exception.

Reply:
Actually I used `null` here for null string(`''`).